osv/purl_helpers.py

```diff
@@ -63,7 +66,84 @@ def package_to_purl(ecosystem: str, package_name: str) -> str | None:
   return f'pkg:{purl_type}/{_url_encode(package_name)}{suffix}'
 
 
-def purl_to_ecosystem(purl_type: str) -> str | None:
-  """Convert purl to a specific ecosystem string"""
-  ecosystem_purl = {v: k for k, v in PURL_ECOSYSTEMS.items()}
-  return ecosystem_purl.get(purl_type)
+def parse_purl(purl_str: str) -> tuple[str, str, str] | None:
```

Comment:
nit: can we used a namedtuple here instead? 

https://docs.python.org/3/library/collections.html#collections.namedtuple

Reply:
Yeah, namedtuple is more suitable for this use case. Updated, thanks!